src/main/java/net/sf/mzmine/modules/peaklistmethods/peakpicking/deconvolution/DeconvolutionTask.java

```diff
@@ -59,6 +61,11 @@
     // User parameters
     private final ParameterSet parameters;
 
+	private RSessionWrapper rSession;

```

Comment:
Would that sort of "Template method pattern" oriented approach please 
you more?

https://github.com/golgauth/mzmine2/tree/rserve-integration-4/src/main/java/net/sf/mzmine/modules/peaklistmethods/peakpicking/deconvolution

This is more or less elegant way of keeping all R-related stuffs 
self-contained in the CentWaveDetector.

(Not yet pushed on the rserve-integration-**4.1** branch)

Readable diff in attachment.


On 05/29/2015 03:05 PM, Tomáš Pluskal wrote:
>
> In 
> src/main/java/net/sf/mzmine/modules/peaklistmethods/peakpicking/deconvolution/DeconvolutionTask.java 
> <https://github.com/mzmine/mzmine2/pull/37#discussion_r31322097>:
>
> > @@ -59,6 +61,11 @@
> >      // User parameters
> >      private final ParameterSet parameters;
> >
> > +	private RSessionWrapper rSession;
>
> If I am not mistaken, CentWave is a singleton itself, instantiated in 
> DeconvolutionParameters:
>
> |     private static final PeakResolver[] RESOLVERS = {
>          new BaselinePeakDetector(), new NoiseAmplitudePeakDetector(),
>          new SavitzkyGolayPeakDetector(), new MinimumSearchPeakDetector(),
>          new CentWaveDetector() };
> |
>
> So although you are right that resolvePeaks() is called many times, 
> the R-related properties could be initialized in the 
> CentWaveDetector's constructor... or not?
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/mzmine/mzmine2/pull/37/files#r31322097>.
>


-- 
Gauthier BOAGLIO
CEFE - UMR 5175
1919 route de Mende
F-34293 Montpellier cedex 5

Tel: +33/0 4 67 61 32 15
Fax: +33/0 4 67 61 33 36

email: gauthier.boaglio@cefe.cnrs.fr
www:   http://www.cefe.cnrs.fr/en/evolutionary-ecology-and-epidemiology/gauthier-boaglio
        http://www.evolepid.org/people.php?name=boaglio